.github/actions/deploy/deploy.py

```
@@ -581,6 +581,32 @@ def deploy_mlflow(self):
+                        if not pair:
+                            continue
+                        if "=" not in pair:
+                            raise ValueError(f"invalid matchLabels pair {pair!r} (expected key=value)")
+                        key, value = pair.split("=", 1)
+                        match_labels[key.strip()] = value.strip()
+                    if not match_labels:
+                        raise ValueError("workspace label selector cannot be empty")
+                    selector_obj = {"matchLabels": match_labels}
+
+                if not isinstance(selector_obj, dict):
+                    raise ValueError("workspace label selector must deserialize to an object")
```

Ruff (0.15.2)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)